src/lava/proc/s4d/models.py

```diff
@@ -0,0 +1,154 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+from lava.proc.sdn.models import AbstractSigmaDeltaModel
+from typing import Any, Dict
+from lava.magma.core.decorator import implements, requires, tag
+import numpy as np
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.proc.s4d.process import SigmaS4Delta, SigmaS4DeltaLayer
+from lava.magma.core.resources import CPU
+from lava.magma.core.model.py.ports import PyInPort, PyOutPort
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.model.sub.model import AbstractSubProcessModel
+from lava.proc.sparse.process import Sparse
+
+
+class AbstractSigmaS4DeltaModel(AbstractSigmaDeltaModel):
+    a_in = None
+    s_out = None
+
+    # SigmaDelta Variables
+    vth = None
+    sigma = None
+    act = None
+    residue = None
+    error = None
+    state_exp = None
+    bias = None
+
+    # S4 Variables
+    A = None
+    B = None
+    C = None
+    S4state = None
+    S4_exp = None
```

Comment:
Python variables should be lowercase. I would also favor using descriptive names over the variable names from the paper.